pkg/overlay/overlay.go

```diff
@@ -61,23 +66,31 @@ func generateOverlayStructure(containerDir string, rootUID, rootGID int) (string
 	return containerDir, nil
 }
 
+// MountWithOptions creates a subdir of the contentDir based on the source directory
+// from the source system.  It then mounts up the source directory on to the
+// generated mount point and returns the mount point to the caller.
+// But allows api to set custom workdir, upperdir and other overlay options
+func MountWithOptions(contentDir, source, dest string, rootUID, rootGID int, graphOptions []string, opts *Options, readonly bool) (mount specs.Mount, Err error) {
```

Comment:
The signature looks inconsistent to me.  Why do we mix `Options` with so arguments? `graphOptions` and `readonly` look optional to me.

Reply:
Yes moved this to options struct.